### PR TITLE
Improved Accuracy for IELTS Success Analysis and Prediction from 0.918990 to 0.999671

### PR DESCRIPTION
**Pull request by DevManpreet5**
# Pull Request for ML-Crate 💡

## Issue Title: Improved Accuracy Comparison for IELTS Success Analysis and Prediction

- **Info about the related issue (Aim of the project)**: To predict the success of IELTS using machine learning models. The existing model used various different models and hyperparameter tuning, increasing performance from XGBoost Regression without tuning (Mean Squared Error: 1.422946e-07, R² Score: 1.000000, Mean Absolute Error: 0.016455, Explained Variance Score: 0.918990) to GradientBoosting Regression with tuning (Mean Squared Error: 1.459225e-06, R² Score: 0.999996, Mean Absolute Error: 0.000067, Explained Variance Score: 0.999671).
- **Name:** Manpreet Singh
- **Email ID for further communication**: singhman2005123@gmail.com
- **GitHub ID:** [devmanpreet5](https://github.com/DevManpreet5)
- **Identify yourself:** Contributing for VSOC 2024

Closes: #663

### Describe the add-ons or changes you've made 📃

I have used various different models and hyperparameter tuning to improve the existing model. Specifically, I improved the performance from XGBoost Regression without tuning (1.422946e-07, 1.000000, 0.016455, 0.918990) to GradientBoosting Regression with tuning (1.459225e-06, 0.999996, 0.000067, 0.999671). The folder containing the updated models, datasets, and results has been added.

## Type of change ☑️

What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

I have tested the changes by dividing the dataset into train and test sets and have checked for over/underfitting. The models were evaluated to ensure performance improvements and robustness.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by abhisheks008** [comment]
You can not contribute in this repository without getting assigned to an issue.

Closing this as not planned.
